src/core/preloadHook.js

```diff
@@ -0,0 +1,44 @@
+const ALLOW_LIST = [
+  "http://localhost.corp.adobe.com:4502" // TODO - Target UI
+];
+
+let preloadListener;
+
+const createResultLogMessage = (origin, success) => {
+  return `Loading script from origin ${origin} ${
+    success ? "succeeded" : "failed"
+  }`;
+};
+
+const preloadHookHandler = resolve => {
+  const preloadHook = event => {
+    if (
+      event.data.type !== "preloadHook" ||
+      !ALLOW_LIST.includes(event.origin)
+    ) {
+      return;
+    }
+
+    try {
+      // eslint-disable-next-line no-eval
+      window.eval(event.data.value);
```

Comment:
This is a big red flag for any security scanner.

Reply:
So there are two overall goals I'm trying to achieve - outlined in this wiki I put together:  https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=Target&title=Alloy+VEC+Integration

The SPA support - which is what uses the monitoring hooks - does use postMessage to send those events to the VEC in the parent window.  That will work fine without any explicit changes to Alloy.  

This PR addresses the second goal - native VEC support.  So basically we need some means of executing our target.js script inside of the iframe before Alloy instances are created.  Given the constraints, I can't think of any way to implement this where Alloy doesn't use eval to run the script.

Reply:
I've updated this to use the reactor-load-script instead of eval

Reply:
Wasn't one of the goals not requiring the customer to allow scripts loaded from target?